### PR TITLE
DB-week3 homework

### PR DESCRIPTION
**Pull request by m-elshawa**
databases week 3 homework

### CODE REVIEW

File: databases/week3/data.sql [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+CREATE DATABASE meal_sharing ;
+USE meal_sharing;
+
+DROP TABLE IF EXISTS Meal;
+CREATE TABLE meal_sharing.Meal (
+    id INT NOT NULL AUTO_INCREMENT,
+    title VARCHAR(45) NOT NULL,
+    description TEXT NOT NULL,
+    location VARCHAR(45) NOT NULL,
+    when DATETIME NOT NULL, -- in my workbench it didn't accept the word when
```
Comment (babakbandpey) [inline databases/week3/data.sql:11]: change the `when `to `time` according to your insert statements below.

File: databases/week3/data2.sql [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+FROM meal
+WHERE price <= 20;
+
+--Get meals that still has available reservations
+
+SELECT meal.title, reservation.created_date AS reservation_date
+FROM meal
+JOIN reservation ON meal.id = reservation.meal_id
+WHERE reservation.created_date > '2020-11-01';
+
```
Comment (babakbandpey) [inline databases/week3/data2.sql:102]: The right answer is;

```
SELECT 
    meal.title,
    reservation.created_date AS reservation_date,
    meal.max_reservations,
    SUM(reservation.number_of_guests)
FROM
    meal
        JOIN
    reservation ON meal.id = reservation.meal_id
WHERE
    reservation.created_date > '2020-11-01'
GROUP BY meal.id , reservation.created_date
HAVING SUM(reservation.number_of_guests) < meal.max_reservations
```

File: databases/week3/data2.sql [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+FROM reservation
+JOIN meal ON reservation.meal_id = meal.id
+WHERE meal.id = 2
+ORDER BY reservation.created_date;
+
+--Sort all meals by average number of stars in the reviews (I don't understand it)
+
+SELECT AVG(review.stars) AS avg_rating, meal.title 
+FROM meal
+LEFT JOIN review ON meal.id = review.meal_id;
```
Comment (babakbandpey) [inline databases/week3/data2.sql:141]: The right answer is
```

SELECT 
    meal.title, round(avg(review.stars)) AS avg_rating
FROM
    meal
        LEFT JOIN
    review ON meal.id = review.meal_id
GROUP BY meal.id
```

**Review by babakbandpey — CHANGES_REQUESTED** [review]
Hello Mohamed

I have commented on some issues.
You may fix them and let me know,
If you have questions please ask, I will try to answer you as best as I can but I expect that this PR is closed soon.
Sincerely Babak

File: databases/week3/data2.sql (hunk 2 above)
Comment (m-elshawa) [reply databases/week3/data2.sql:102]: But why do I need to do all of that if the required is only to get the meals with available reservations? 
What is wrong with my code? (just asking to understand)

Comment (m-elshawa) [reply databases/week3/data2.sql:102]: OK sorry, apparently I totally miss-understood the question, but now I get it. Thanks for your feedback :)

File: databases/week3/data2.sql (hunk 3 above)
Comment (m-elshawa) [reply databases/week3/data2.sql:141]: Now I got it. Thanks man :)

**Pull request merged**